client.go

```diff
@@ -809,7 +809,7 @@ func (c *Client[TTx]) distributeJobCompleterCallback(update jobcompleter.Complet
 		c.statsNumJobs++
 	}()
 
-	c.distributeJob(jobRowFromInternal(update.Job), jobStatisticsFromInternal(update.JobStats))
+	c.distributeJob((*JobRow)(dbsqlc.JobRowFromInternal(update.Job)), jobStatisticsFromInternal(update.JobStats))
```

Comment:
Not that all of these currently convert `river.JobRow` <-> `dbsqlc.RiverJob`, but in the future we'd expect `dbsqlc` to get pushed entirely into the driver.

(`dbsqlc` is pgx v5 specific, so if we were to support an alternate package, it'd need its own dbsqlc-like package. For example, if we supported pgx v4, we'd have a dbsqlc configured to emit pgx v4, and that'd live in the `riverpgxv4` driver.)

So the driver would convert to and return a `rivertype.JobRow`, and then all these conversion would become simple `(*JobRow)(update.Job)`.